File: src/main/java/cs/hse/springWebJavaDemo/features/songs/controllers/AlbumController.java
package cs.hse.springWebJavaDemo.features.songs.controllers;

import cs.hse.springWebJavaDemo.features.songs.dto.AlbumDto;
import cs.hse.springWebJavaDemo.features.songs.services.AlbumService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/music/albums")
@RequiredArgsConstructor
class AlbumController {

    private final AlbumService albumService;

    @GetMapping
    ResponseEntity<List<AlbumDto>> getAllAlbums()  {
        return ResponseEntity.ok(albumService.retrieveAlbums());
    }

    @GetMapping("/get")
    ResponseEntity<AlbumDto> getAlbumWithId(@RequestParam String id) {
        var album = albumService.retrieveAlbumById(id);
        return ResponseEntity.ok(album);

    }

    @PostMapping("/add")
    ResponseEntity<String> postAnAlbum(@RequestBody AlbumDto album) {
        var resp = albumService.tryAddAlbum(album);
        if (resp.contains("ERROR")) {
            ResponseEntity.badRequest().body(resp);
        }

        return ResponseEntity.ok(resp);
    }
}

File: src/main/kotlin/cs/hse/springWebDemo/features/songs/controllers/AlbumController.kt
package cs.hse.springWebDemo.features.songs.controllers

import cs.hse.springWebDemo.features.songs.dto.AlbumDto
import cs.hse.springWebDemo.features.songs.services.AlbumService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/music/albums")
class AlbumController (private val albumService: AlbumService) {

    @GetMapping
    fun getAllAlbums() : ResponseEntity<List<AlbumDto>> {
        return ResponseEntity.ok(albumService.retrieveAlbums())
    }

    @GetMapping("/get/{id}")
    fun getAlbumWithId(@PathVariable id: Long) : ResponseEntity<AlbumDto> {
        val album = albumService.retrieveAlbumById(id) ?: return ResponseEntity.notFound().build()
        return ResponseEntity.ok(album)
    }

    @GetMapping("/get")
    fun getAlbumsByArtist(@RequestParam artist: String) : ResponseEntity<List<AlbumDto>> {
        val albums = albumService.retrieveAlbumsByArtist(artist)
        return ResponseEntity.ok(albums)
    }

    @PostMapping("/add")
    fun postAnAlbum(@RequestBody album: AlbumDto) : ResponseEntity<String> {
        val resp = albumService.tryAddAlbum(album) ?: return ResponseEntity.badRequest().build()
        return ResponseEntity.ok(resp.toString())
    }
}